### PR TITLE
Feature/login page

### PR DESCRIPTION
**Pull request by pnarozniak**
- Added reusable components
- Added login page

### CODE REVIEW

File: Client/Components/ExternalServicesComponent.razor.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    font-family: 'Roboto', sans-serif;
+    font-size: 20px;
+    font-weight: bold;
+    height: 67px;
+    margin-bottom: 1.6rem;
+    outline: none;
+    width: 401px;
+}
+
+    button img {
```
Comment (Ventan00) [inline Client/Components/ExternalServicesComponent.razor.css:15]: wrong code formating

File: Client/Components/FormSubmitComponent.razor.css [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    font-size: 20px;
+    font-weight: bold;
+    height: 57px;
+    margin: 0.8rem 0;
+    outline: none;
+    transition: letter-spacing 300ms ease-in;
+    width: 100%;
+}
+
+    .form-submit-btn:hover {
```
Comment (Ventan00) [inline Client/Components/FormSubmitComponent.razor.css:16]: wrong code formatting

File: Client/Components/TopSemicircleComponent.razor.css [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿.img-container {
+    height: 160px;
+    overflow: hidden;
+    position: relative;
+}
+
+    .img-container img {
```
Comment (Ventan00) [inline Client/Components/TopSemicircleComponent.razor.css:7]: wrong code formatting

File: Client/Components/LoginFormComponent.razor [hunk 4]
```diff
@@ -46,40 +26,26 @@
...
-        }
-        _loginDto.CaptchaResponseToken = captchaResponseToken;
-
-        _loginStatus = null;
         _isPending = true;
-        StateHasChanged();
+        _loginDto.CaptchaResponseToken = await _form.CaptchaComponent.GetCaptchaResponseAsync();
 
         var loginResponseDto = await _authenticationService.LoginAsync(_loginDto);
-        await HandleLoginResponse(loginResponseDto.LoginStatus, loginResponseDto.Avatar);
```
Comment (Ventan00) [inline Client/Components/LoginFormComponent.razor:82]: Why did you remove avatar?

File: Client/Components/LoginFormComponent.razor [hunk 5]
```diff
@@ -88,32 +54,30 @@
...
-    private async Task HandleLoginResponse(LoginStatusEnum loginStatusEnum, string Avatar)
+    private async Task HandleLoginResponse(LoginStatusEnum loginStatusEnum)
     {
-        _loginStatus = loginStatusEnum;
-
         switch (loginStatusEnum)
         {
             case LoginStatusEnum.LOGIN_ACCEPTED:
                 _toastService.ShowSuccess("You have logged in.");
-                await _localStorageService.SetItemAsync("avatar", Avatar);
```
Comment (Ventan00) [inline Client/Components/LoginFormComponent.razor:102]: Why did you remove saving avatar to local storage?

**Review by Ventan00 — CHANGES_REQUESTED** [review]
Wrong code styling.
Need explaination why avatar transfering functionality was removed.

File: Client/Components/ExternalServicesComponent.razor.css (hunk 1 above)
Comment (xlegodroit) [reply Client/Components/ExternalServicesComponent.razor.css:15]: In my opionion this formating is correct. It should be tabbed right because it makes code more redeable. Because you modify img inside button so its tabbed right.

**Comment by pnarozniak** [comment]
In my opionion this formating is correct. It should be tabbed right. That's how you write css in my opinion because it makes code more redeable. Because you style child inside it's parrent(over him) so its tabbed right.

**Comment by pnarozniak** [comment]
I copied LoginFormComponent.razor from old version of develop branch so that's why it is removed.  I'll fix it

**Review by Patrykoss — APPROVED** [review]
For me it's ok with that type of code stylling in css.

**Review by pnarozniak — COMMENTED** [review]
I modified param from loginStatusEnum, avatar to LoginResponseDto it will make code more redeable if we will extend this dto and need to add logic to this method in the future.

**Review by Ventan00 — APPROVED** [review]
Now OK. From now on please use this kind of css formatting everywhere in project.
